### ISSUE TITLE
08-k-nearest-neighbours-winequality-red.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/machine-learning-elius123ef#6]
Hola @elius123ef , por aquí las issues!

## LO QUE HAS HECHO MUY BIEN

- **Eliminación de duplicados correcta**: `df = df.drop_duplicates()`.

- **Análisis de nulos y duplicados**: Verificaste ambos.

- **Heatmap de correlaciones**: Bien visualizado con observaciones.

- **Observaciones de correlaciones**: Excelente análisis.

- **Boxplots para outliers**: Buena visualización.

- **Creación de variable de calidad**: Correcto con `pd.cut()`.

- **Escalado con ColumnTransformer**: Muy profesional usar `ColumnTransformer` con `StandardScaler`. ¡Y lo usaste correctamente!

- **TODAS las métricas pedidas**

- **Observaciones de matriz de confusión**: Excelente análisis del sesgo hacia clase mayoritaria.

- **Observaciones del classification_report**: Muy buen análisis con porcentajes de cada clase.

- **Bucle de k + gráfica**: ¡PERFECTO! Probaste k de 1 a 20 y graficaste accuracy vs k.

- **Comparación GridSearchCV vs RandomizedSearchCV**: Muy completo probar ambos métodos.

- **Modelo final con mejores parámetros**: Combinaste el mejor k encontrado con otros hiperparámetros.

---

### **1. GridSearchCV y RandomizedSearchCV con datos SIN escalar**
```python
grid_search.fit(X_train, y_train)  # ← SIN escalar
random_search.fit(X_train, y_train)  # ← SIN escalar
```

Pero el modelo final sí usa datos escalados:
```python
model.fit(X_train_ready, y_train)  # ← Escalado
```

Esto hace que las comparaciones no sean justas. Deberías usar `X_train_ready` en todas las búsquedas.

---

### **2. NO guardaste el modelo**
No hay código con pickle para guardar el modelo final. Importaste `dump` pero no lo usaste.

---

### **3. Imports innecesarios**
Tienes muchos imports que no usas en este ejercicio:
- `WordCloud`
- `fetch_20newsgroups`
- `TfidfVectorizer`, `CountVectorizer`
- `GaussianNB`, `MultinomialNB`, `BernoulliNB`
- `Pipeline`

Estos son del ejercicio de Naive Bayes, no de KNN. Mejor dejar menos pesado y limpio tu trabajo

---

**Muy buen trabajo, Elius.** Tu análisis es muy completo y las observaciones demuestran que entiendes bien el problema del desbalance de clases. El único tema importante es la inconsistencia en el escalado durante las búsquedas de hiperparámetros y te faltó guardar el modelo.
